Sources/RDeviceIdentifier/RDeviceIdentifier.swift

```diff
@@ -40,47 +40,26 @@ import RLogger // Required in SPM version
     ///         If the keychain is not available (i.e. the device is locked) and no value has been
     ///         retrieved yet, `nil` is returned and the developer should try again when the device
     ///         is unlocked.
-    @objc public static var uniqueDeviceIdentifier: String? {
-        // If we already have a value, return it.
-        if let _uniqueDeviceIdentifier = _uniqueDeviceIdentifier {
-            return _uniqueDeviceIdentifier
-        }
-
-        var accessGroup: String?
-        do {
-            if let (internalAccessGroup, defaultAccessGroup) = try keychain.getAccessGroups() {
-                accessGroup = internalAccessGroup
-                checkKeychainAccessGroupOrder(defaultAccessGroup)
-            }
-        } catch {
-            // Keychain is not available
-            RLogger.debug(message: "Keychain access failed")
-            return nil
-        }
-
-        // Try to clean things up
-        reset()
+    public static func getUniqueDeviceIdentifier(service: String, accessGroup: String) throws -> String {
+        let keychain = RDeviceIdentifierKeychain(service: service, accessGroup: accessGroup)
```

Comment:
In this new architecture, is it possible to inject the keychain in an internal function `getUniqueDeviceIdentifier` that would be:
`public static func getUniqueDeviceIdentifier(service: String, accessGroup: String, keychain: RDeviceIdentifierKeychainable) throws -> String`
Where RDeviceIdentifierKeychainable is a protocol that RDeviceIdentifierKeychain implements.
?

So we can use this function to inject a mock of RDeviceIdentifierKeychain in the target tests.